Behavioral.Automation.API/Bindings/HttpResponseSteps.cs

```diff
@@ -0,0 +1,289 @@
+﻿using System.Text.RegularExpressions;
+using Behavioral.Automation.API.Context;
+using Behavioral.Automation.API.Models;
+using Behavioral.Automation.API.Services;
+using Behavioral.Automation.Configs.utils;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+using Polly;
+using TechTalk.SpecFlow;
+
+namespace Behavioral.Automation.API.Bindings;
+
+[Binding]
+public class HttpResponseSteps
+{
+    private readonly ApiContext _apiContext;
+    private readonly HttpService _httpService;
+    private readonly int _retryCount = 10;
+    private readonly TimeSpan _retryDelay = TimeSpan.FromSeconds(3);
+
+    public HttpResponseSteps(ApiContext apiContext, HttpService httpService)
+    {
+        _apiContext = apiContext;
+        _httpService = httpService;
+    }
+
+    [Then("response attachment filename is \"(.*)\"")]
+    public void ThenResponseAttachmentFilenameIs(string filename)
+    {
+        if (_apiContext.Response is null) throw new NullReferenceException("Http response is empty.");
```

Comment:
In this case, NullReferenceException throws if an object is null so I think it is fine. The root cause of the issue is that the request wasn't sent so it more test consistency issue.

Reply:
In C# `NullReferenceException` is generally thrown when user tries to access `null` instead of valid value. Here no one tries to access it, so logically it seems right to use other kind, maybe not `NullReferenceException`, but definitely not `NullReferenceException`

Reply:
Changed.